include/aws/http/request_response.h

```diff
@@ -234,6 +270,13 @@ struct aws_http_make_request_options {
      */
     aws_http_on_incoming_body_fn *on_response_body;
 
+    /**
+     * Invoked right before stream is complete, whether successful or unsuccessful
+     * Optional.
+     * See `aws_http_on_stream_metrics_fn`
+     */
+    aws_http_on_stream_metrics_fn *on_metrics;
```

Comment:
[debatable] Would it be simpler to have a query function, instead of a callback? similar existing query functions such as aws_http_stream_get_incoming_response_status(...)

```
// data not available until the stream is complete
const struct aws_http_stream_metrics *aws_http_stream_get_metrics(const struct aws_http_stream *stream);
```

Reply:
I think if we use callbacks, we can skip gathering metrics if the callback was not set, which can improve the performance.